### PR TITLE
Feature/20 qmd headers support

### PR DESCRIPTION
**Pull request by ran-codes**
This pull request adds comprehensive support for detecting header sections in Markdown and Quarto files, including logic to ignore headers inside code blocks. It also updates all relevant tests to use the new language-specific parsing for section detection, ensuring more accurate and robust results across file types.

**Markdown and Quarto Section Detection Improvements:**

* Enhanced `findSections` in `src/utils/findSections.ts` to detect Markdown and Quarto headers as sections, even without trailing dashes, and to ignore headers found inside code blocks. [[1]](diffhunk://#diff-7534e6c0619cc989d8b105483dd5720ea8797793000538467775506a8b1d2357R15-R64) [[2]](diffhunk://#diff-7534e6c0619cc989d8b105483dd5720ea8797793000538467775506a8b1d2357L46-R92) [[3]](diffhunk://#diff-7534e6c0619cc989d8b105483dd5720ea8797793000538467775506a8b1d2357L62-R109)
* Added new test suite `src/test/md-comments.test.ts` to verify section detection in realistic Markdown documents, including nested headers and edge cases.
* Added new test suite `src/test/quarto-comments.test.ts` to ensure Quarto header sections are detected, and headers inside code blocks are ignored.

**Test Updates for Language-specific Parsing:**

* Updated all relevant tests in `src/test/jsx-comments.test.ts` to pass the correct language identifier (`'jsx'` or `'tsx'`) to `findSections`, ensuring proper parsing logic is applied for each file type. [[1]](diffhunk://#diff-447c475826f03c35023aeadd56332aaaecfd8c38bdb845b228d8fe4391981c4dL26-R26) [[2]](diffhunk://#diff-447c475826f03c35023aeadd56332aaaecfd8c38bdb845b228d8fe4391981c4dL65-L87) [[3]](diffhunk://#diff-447c475826f03c35023aeadd56332aaaecfd8c38bdb845b228d8fe4391981c4dL101-R94) [[4]](diffhunk://#diff-447c475826f03c35023aeadd56332aaaecfd8c38bdb845b228d8fe4391981c4dL116-R108) [[5]](diffhunk://#diff-447c475826f03c35023aeadd56332aaaecfd8c38bdb845b228d8fe4391981c4dL132-R123) [[6]](diffhunk://#diff-447c475826f03c35023aeadd56332aaaecfd8c38bdb845b228d8fe4391981c4dL153-R143) [[7]](diffhunk://#diff-447c475826f03c35023aeadd56332aaaecfd8c38bdb845b228d8fe4391981c4dL188-R177)

### CODE REVIEW

File: src/utils/findSections.ts [hunk 1]
```diff
@@ -12,14 +12,56 @@ export interface SectionMatch {
...
+
+  // For Markdown/Quarto files, find code block ranges to exclude from parsing
+  const codeBlocks: { start: number; end: number }[] = [];
+  if (isMarkdownOrQuarto) {
+    const lines = text.split('\n');
+    let inCodeBlock = false;
+    let codeBlockStart = 0;
+    
+    lines.forEach((line, index) => {
+      if (line.trim().startsWith('```')) {
```
[the remark this answers is not in the document]
Comment (ran-codes) [reply src/utils/findSections.ts:32]: thanks fixed in https://github.com/ran-codes/code-organizer-vscode/commit/11352c0ade431d019be616b09bbe6cb8530e7995

File: src/utils/findSections.ts [hunk 2]
```diff
@@ -12,14 +12,60 @@ export interface SectionMatch {
...
+      codeBlocks.push({ start: codeBlockStart, end: lines.length - 1 });
+    }
+  }
+
+  // Helper function to check if a match index is inside a code block
+  const isInCodeBlock = (matchIndex: number): boolean => {
+    if (!isMarkdownOrQuarto) return false;
+    
+    const lines = text.substring(0, matchIndex).split('\n');
+    const matchLineNumber = lines.length - 1;
```
[the remark this answers is not in the document]
Comment (ran-codes) [reply src/utils/findSections.ts]: resoveld

File: src/utils/findSections.ts [hunk 3]
```diff
@@ -12,14 +12,72 @@ export interface SectionMatch {
...
+  let lineStartIndices: number[] = [];
+  if (isMarkdownOrQuarto) {
+    const lines = text.split('\n');
+    let inCodeBlock = false;
+    let codeBlockStart = 0;
+    let charIndex = 0;
+    lineStartIndices = [];
+    lines.forEach((line, index) => {
+      lineStartIndices.push(charIndex);
+      if (line.trim().startsWith('```')) {
```
[the remark this answers is not in the document]
Comment (ran-codes) [reply src/utils/findSections.ts:32]: reverting
